annotator/src/main/java/org/cbioportal/annotator/util/AnnotationUtil.java

```diff
@@ -331,11 +331,18 @@ public String resolveHgvspShort(TranscriptConsequenceSummary canonicalTranscript
         return hgvsp;
     }
 
-    public String resolveConsequence(TranscriptConsequenceSummary canonicalTranscript) {
-        if (canonicalTranscript == null || canonicalTranscript.getConsequenceTerms() == null) {
-            return "";
-        } else {
+    public String resolveConsequence(VariantAnnotation gnResponse, TranscriptConsequenceSummary canonicalTranscript) {
+        if (canonicalTranscript != null && canonicalTranscript.getConsequenceTerms() != null) {
             return canonicalTranscript.getConsequenceTerms();
+        } else if (gnResponse != null && 
+                   gnResponse.getAnnotationSummary() != null && 
+                   gnResponse.getAnnotationSummary().getIntergenicConsequenceSummaries() != null &&
+                   !gnResponse.getAnnotationSummary().getIntergenicConsequenceSummaries().isEmpty()) {

```

Comment:
Should we move this to a method like `hasIntergenicConsequenceSummaries` and reuse in the other method as well?

Reply:
Moved it out to a separate util funcion